protocols/direct-fund/direct-fund.go

```diff
@@ -179,6 +179,12 @@ func (s DirectFundObjective) Crank(secretKey *[]byte) (protocols.Objective, prot
 	return updated, NoSideEffects, WaitingForNothing, nil
 }
 
+func (s DirectFundObjective) Channels() []types.Destination {
+	ret := make([]types.Destination, 0, 1)
+	ret = append(ret, s.C.Id)
```

Comment:
1. We can append as many values as we like, but the array which backs the slice has a pre-allocated length of 1. If we append more than one value, the array size is dynamically adjusted (doubled, or something, and maybe moved to new physical location in memory etc). Defining `capacity` of `n` is good practice if you know in advance how large your slice will be, or an upper bound on that size, since it precludes these memory reallocation operations.

The rest looks correct up to my understanding.